pixels-planner/src/main/java/io/pixelsdb/pixels/planner/coordinate/StageCoordinator.java

```diff
@@ -102,12 +102,15 @@ public StageCoordinator(int stageId, List<Task> pendingTasks)
      */
     public void addWorker(Worker<CFWorkerInfo> worker)
     {
-        this.workerIdToWorkers.put(worker.getWorkerId(), worker);
-        this.workerIdToWorkerIndex.put(worker.getWorkerId(), this.workerIndexAssigner.getAndIncrement());
-        this.workers.add(worker);
-        if (!this.isQueued && this.workers.size() == this.fixedWorkerNum)
+        synchronized (this.lock)
```

Comment:
Why synchronize on lock?

Reply:
Previously when we did not have this `synchronized`, I got an exception `java.lang.IllegalMonitorStateException: current thread is not owner`. Google and ChatGPT gave me this solution.

Reply:
For more info, I called `WorkerCoordinateService.registerWorker()` in each worker, which in turn calls into here, and obviously there are multiple workers running in their respective thread.

Reply:
Siyuan also modified this method. Please discuss this with him to avoid conflicts.